src/service/viewport-impl.js

```diff
@@ -302,6 +304,29 @@ export class Viewport {
   }
 
   /**
+   * Scrolls element into view much like Element. scrollIntoView does but
+   * in the AMP/Viewer environment. Adds animation for the sccrollIntoView
+   * transition.
+   *
+   * @param {!Element} element
+   * @param {number=} duration
+   * @param {string=} curve
+   * @return {!../../animation.AnimationPlayer}
+   */
+  animateScrollIntoView(element, duration = 500, curve = 'ease-in') {
+    const elementTop = this.binding_.getLayoutRect(element).top;
+    const newScrollTop = Math.max(0, elementTop - this.paddingTop_);
+    const interpolate = numeric(this.getScrollTop(), newScrollTop);
+    // TODO(erwinm): the duration should not be a constant and should
+    // be done in steps for better transition experience when things
+    // are closer vs farther.
+    return Animation.animate(pos => {
+      pos = interpolate(pos);
```

Comment:
`pos = interpolate(pos)` freaks me out a bit since these values are in different coord systems (one could say even types). Could you please just combine these two lines?